requests/adapters.py

```diff
@@ -284,7 +302,13 @@ def request_url(self, request, proxies):
         """
         proxy = select_proxy(request.url, proxies)
         scheme = urlparse(request.url).scheme
-        if proxy and scheme != 'https':
+        if proxy:
+            proxy_scheme = urlparse(proxy).scheme
+
+        if proxy and proxy_scheme.lower().startswith('socks'):
```

Comment:
I'm being a little tricksy here, because `and` is short-circuiting in Python. This means that I actually only reference `proxy_scheme` here if `proxy` is truthy, which is the condition under which we populate it in the first place.

However, it's definitely a bit unclear, so I'm happy to change it.